### DISCUSSION TITLE
Photo download API

### DISCUSSION THREAD

**Discussion by Atreb92** [photoprism/photoprism#3726] [Q&A] [answered]
Hello,
I was playing around with photoprism APIs for a side project I am working on and I noticed that the API in charge of downloading an image (/api/v1/dl/:hash) allows photo download without any authentication.

Is photoprism relying on the fact that the SHA1 (which is used as photo identifier) total value range would require an unfeasable amount of time in order to hit any valid values?

Thank you for the clarification!

**Comment by lastzero** [comment, marked as answer]
It should actually expect a download token as parameter, unless you are running your instance in public mode?

**Reply by Atreb92** [reply]
Hi, thanks for the reply!
No, I just checked the docker config and I am running password mode. Is the token you are referring to the URL parameter "t"?

**Reply by lastzero** [reply]
Yes, exactly 👍

**Reply by Atreb92** [reply]
Oh ok, makes sense. Thank you for the clarification! Is this token retrievable by the APIs?

**Reply by lastzero** [reply]
You should get it when you create a session. If it changes during the session, the updated token is submitted in the header of responses you get from the server, so your client can update it from there.

**Reply by lastzero** [reply]
Should be X-Download-Token if I remember correctly. On my phone right now, so I can't look it up in the source. You'll find it there.

**Reply by Atreb92** [reply]
Thanks for the information, will look into it!
